src/host/random.jl

```diff
@@ -84,29 +84,32 @@ function Random.seed!(rng::RNG, seed::Vector{UInt32})
 end
 
 function Random.rand!(rng::RNG, A::AnyGPUArray{T}) where T <: Number
-    gpu_call(A, rng.state) do ctx, a, randstates
-        idx = linear_index(ctx)
-        idx > length(a) && return
+    @kernel rand!(a, randstate)
+        idx = @index(Global, Linear)
         @inbounds a[idx] = gpu_rand(T, ctx, randstates)
```

Comment:
Is this the only way people will be using `gpu_rand()`? Or will they be calling the functions on lines {33, 40, 58...}?

If so, we can just pass `idx` as `threadid` into the above functions, right?

Reply:
The only supported API are the Random stdlib functions; calling `gpu_rand` explicitly is not supported, so feel free to refactor.